subsys/bluetooth/audio/bap_stream.c

```diff
@@ -78,6 +78,23 @@ void bt_audio_codec_qos_to_iso_qos(struct bt_iso_chan_io_qos *io,
 	* CONFIG_BT_BAP_BROADCAST_SINK                                                             \
 	*/
 
+void bt_bap_stream_init(struct bt_bap_stream *stream)
+{
+	stream->dir = 0x00;
+	stream->conn = NULL;
```

Comment:
No, the object is allocated by the user. How would you know whether `stream->conn` points to valid `conn` object??

Reply:
We don't, and that's the issue with this whole user-owned-stack-controlled way that L2CAP, ISO, BAP (and now also) CAP streams are defined. 


The question thus: If `stream->conn != NULL`, and returned by the user, should we assume that
1) The stream object is already in use, in which case we should reject it here
2) That the user did not provide a initialized stream (e.g. if `stream->conn` is uninitialized) 
3) That the user themselves did `stream->conn = bt_conn_ref(...)`? 

Given that we can only assume, I'd recommend going with the safer route and reject any stream where `stream->conn != NULL`, together with `codec`, `qos` and `bap_iso`. All of these values are (or should be) set to NULL by the stack. (I see though, that for the server, only `bt_bap_stream_detach` is called, which does not set `qos` to NULL)

Reply:
The fact that the stream object is not opaque is to allow the user allocate it. It is understandable that if the user will modify `stream` object members it may lead to unexpected undesirable behavior.
If you are worried about `conn` to be referenced, the solution would be to change the `bt_bap_unicast_server_cb.config` `conn` parameter to be `const`.

Reply:
int (*config)(**const** struct bt_conn *conn, const struct bt_bap_ep *ep, enum bt_audio_dir dir,
		      const struct bt_codec *codec, struct bt_bap_stream **stream,
		      struct bt_codec_qos_pref *const pref);

Reply:
IMO making `struct bt_conn *conn` const, doesn't change anything here, it only gives an error if the upper layer does something like `stream->conn = conn`, but it doesn't stop the upper laying from doing `stream->conn = other_conn` (or getting a non-const point with the same connection using e.g. `bt_conn_foreach`).